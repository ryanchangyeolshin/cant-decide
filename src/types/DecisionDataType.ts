type DecisionDataType = {
  id?: string,
  user: string,
  name: string,
  choices: string[],
};

export default DecisionDataType;