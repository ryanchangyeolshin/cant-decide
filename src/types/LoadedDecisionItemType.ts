import React from 'react';
import DecisionDataType from './DecisionDataType';

type LoadedDecisionItemType = {
  decision: DecisionDataType,
  setSelectedDecisionHandler: (e: React.MouseEvent<HTMLElement>, decision: DecisionDataType) => void
};

export default LoadedDecisionItemType;