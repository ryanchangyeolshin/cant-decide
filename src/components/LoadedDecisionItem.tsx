import React from 'react';
import { makeStyles, createStyles, Theme, } from '@material-ui/core/styles';
import {
  ListItem,
  ListItemText,
  Grow
} from '@material-ui/core';
import LoadedDecisionItemType from '../types/LoadedDecisionItemType';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    listItem: {
      background: '#F4ECDA',
      color: "black",
      marginBottom: "5px",
      borderRadius: "4px",
      padding: "20px",
      transition: theme.transitions.create("background", {
        duration: 1000
      }),
      "&:focus": {
        background: "yellow",
      },
      "& .MuiListItemText-root": {
        "& .MuiTypography-root.MuiListItemText-primary": {
          fontWeight: 900,
          overflow: "hidden",
          whiteSpace: "nowrap",
          textOverFlow: "ellipsis",
          width: "95%"
        }
      },
      "& .MuiBadge-colorPrimary": {
        background: "black"
      }
    }
  }),
);

const LoadedDecisionItem: React.FC<LoadedDecisionItemType> = ({ decision, setSelectedDecisionHandler }: LoadedDecisionItemType) => {
  const classes = useStyles();

  return (
    <Grow in={true} timeout={1500} mountOnEnter unmountOnExit={true}>
      <div
        className="loaded-decision-item-container"
        onClick={(e: React.MouseEvent<HTMLElement>) => setSelectedDecisionHandler(e, decision)}
      >
        <ListItem button className={classes.listItem}>
          <ListItemText id="decision-text" primary={decision.name} />
        </ListItem>
      </div>
    </Grow>
  );
}

export default LoadedDecisionItem;