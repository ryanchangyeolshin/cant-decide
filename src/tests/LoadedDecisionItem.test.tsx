import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import LoadedDecisionItem from '../components/LoadedDecisionItem';
import LoadedDecisionItemType from '../types/LoadedDecisionItemType';
import DecisionDataType from '../types/DecisionDataType';

afterEach(cleanup);

describe("This suit is to test the LoadedDecisionItem component", () => {
  const props: LoadedDecisionItemType = {
    decision: { _id: "192e8du1id", name: "What to eat?!", user: "johndoe@gmail.com", choices: ["Taco Bell", "KFC"] },
    setSelectedDecisionHandler: jest.fn((e: React.MouseEvent<HTMLElement>, decision: DecisionDataType) => "Decision Selected!")
  };

  test('Snapshot of LoadDecisionModal', () => {
    const { asFragment } = render(
      <LoadedDecisionItem
        decision={props.decision}
        setSelectedDecisionHandler={props.setSelectedDecisionHandler}
      />
    );
    expect(asFragment()).toMatchSnapshot()
  });

  test('that the login button is rendered', async () => {
    const { container } = await render(
      <LoadedDecisionItem
        decision={props.decision}
        setSelectedDecisionHandler={props.setSelectedDecisionHandler}
      />
    );
    const decisionText = container.querySelector('#decision-text') as HTMLElement;
    expect(decisionText.textContent).toBe(props.decision.name);
  });

  test('that the setSelectedDecisionHandler is called', async () => {
    const { container } = await render(
      <LoadedDecisionItem
        decision={props.decision}
        setSelectedDecisionHandler={props.setSelectedDecisionHandler}
      />
    );
    const loadedDecisionItemContainer = container.querySelector('.loaded-decision-item-container') as HTMLElement;
    fireEvent.click(loadedDecisionItemContainer)
    expect(props.setSelectedDecisionHandler).toBeCalled();
  });
});
