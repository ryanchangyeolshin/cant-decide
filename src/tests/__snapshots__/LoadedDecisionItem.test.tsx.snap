// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the LoadedDecisionItem component Snapshot of LoadDecisionModal 1`] = `
<DocumentFragment>
  <div
    class="loaded-decision-item-container"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 999ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root"
        id="decision-text"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          What to eat?!
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
